pkg/manager/router/backend_observer.go

```diff
@@ -178,13 +180,26 @@ func (bo *BackendObserver) Refresh() {
 }
 
 func (bo *BackendObserver) observe(ctx context.Context) {
+	var lastErr error
 	for ctx.Err() == nil {
-		backendInfo := bo.fetcher.GetBackendList(ctx)
-		backendStatus := bo.checkHealth(ctx, backendInfo)
-		if ctx.Err() != nil {
-			return
+		backendInfo, err := bo.fetcher.GetBackendList(ctx)
+		if err == nil {
+			if lastErr != nil {
+				bo.logger.Info("fetching backends succeeds now")
+				bo.eventReceiver.OnObserveError(err)
```

Comment:
How about use only callback `OnObserve(map[string]XXX, error)` or `OnObserve(ObserveEvent)`?
I feel it is a little bit complicated that `backendChangeEvent` and `backendChangeErrorEvent` are parallel.

It seems more reasonable to invoke `OnXXXEvent` everytime:
1. polled the fetcher, and something is returned
2. gateway asynchronizely pushed some events to the observer